drivers/sensor/tmd2620/Kconfig

```diff
@@ -0,0 +1,122 @@
+# Copyright (c) 2022 PHYTEC Messtechnik GmbH
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig TMD2620
+	bool "OSRAM-AMS TMD2620 Proxmimity Sensor"
+	default y
+	depends on DT_HAS_AMS_TMD2620_ENABLED
+	select I2C
+	help
+	  Enable driver for TMD2620 Sensor
+
+if TMD2620
+
+choice
+	prompt "Trigger Mode"
+	default TMD2620_TRIGGER_NONE
+	help
+	  Specify the type of triggering to be used by the driver.
+
+config TMD2620_TRIGGER_NONE
+	bool "No Trigger"
+
+config TMD2620_TRIGGER_GLOBAL_THREAD
+	bool "Use global thread"
+	depends on GPIO
+	select TMD2620_TRIGGER
+
+endchoice # Trigger Mode
+
+config TMD2620_TRIGGER
+	bool
+
+choice
+	prompt "Proxmity Gain"
+	default TMD2620_PGAIN_X4
+
+config TMD2620_PGAIN_X1
+	bool "1x"
+
+config TMD2620_PGAIN_X2
+	bool "2x"
+
+config TMD2620_PGAIN_X4
+	bool "4x"
+
+config TMD2620_PGAIN_X8
+	bool "8x"
+
+endchoice # Proxmity Gain
+
+choice
+	prompt "Proximity Pulse Length"
+	default TMD2620_PPULSE_LEN_16US
+
+config TMD2620_PPULSE_LEN_4US
+	bool "4us"
+
+config TMD2620_PPULSE_LEN_8US
+	bool "8us"
+
+config TMD2620_PPULSE_LEN_16US
+	bool "16us"
+
+config TMD2620_PPULSE_LEN_32US
+	bool "32us"
+
+endchoice # Proximity Pulse Length
+
+config TMD2620_PULSE_COUNT
+	int "Proximity Pulse Count"
+	default 15
+	range 1 64 # decrement by 1
+
+config TMD2620_LED_DRIVE_STRENGTH
+	int "Proximity LED Drive Strength (x * 6mA)"
+	default 7
+	range 1 32 # decrement by 1
+
+config TMD2620_PILT
+	int "Proximity Interrupt Low Threshold"
+	default 0
+	range 0 255
+
+config TMD2620_PIHT
+	int "Proximity Interrupt High Threshold"
+	default 1 if !TMD2620_TRIGGER_GLOBAL_THREAD
+	default 127 if TMD2620_TRIGGER_GLOBAL_THREAD
+	range 0 255
+
+config TMD2620_PPERS
+	int "Proximity Interrupt Filter"
+	help
+		0: every proximity cycle generates an interrupt
+		1: 1 consecutive proximity value out of threshold range
+		2: 2 consecutive proximity values out of threshold range
+		...
+	default 0 if !TMD2620_TRIGGER_GLOBAL_THREAD
+	default 7 if TMD2620_TRIGGER_GLOBAL_THREAD
+	range 0 15
+
+config TMD2620_WAIT_ENABLE
+	bool "Enable wait mode"
+	default n
+
+config TMD2620_WTIME
+	int "Wait Time"
+	help
+		0: disable wait timer
+		1: 2.78ms
+		2: 5.56ms
+		...
+	default 255
+	range 1 256 # decrement by 1
+
+config TMD2620_WLONG
+	bool "Wait Long"
+	help
+		increase wait time by a factor of 12x
+	default n
```

Comment:
Please convert these to devicetree properties to allow per-instance configuration